datalake/backend/tests/test_processing_stack.py

```diff
@@ -0,0 +1,113 @@
+
+from pytest import mark
+
+APPLICATION_TAG_KEY = "ApplicationName"
+APPLICATION_TAG_VAL = "geospatial-data-lake"
+
+ENVIRONMENT_TAG_KEY = "EnvironmentType"
+ENVIRONMENT_TAG_VAL = os.environ.get("DEPLOY_ENV", "dev")
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def get_compute_environment(batch_client):
+    compute_environment_detection_response = batch_client.describe_compute_environments()
+
+    # make sure there are no more records in next pages
+    assert compute_environment_detection_response.get("nextToken", None) is None
+
+    for compute_environment in compute_environment_detection_response["computeEnvironments"]:
+        if (
+            compute_environment["tags"][ENVIRONMENT_TAG_KEY] == ENVIRONMENT_TAG_VAL
+            and compute_environment["tags"].get(APPLICATION_TAG_KEY, None) == APPLICATION_TAG_VAL
+        ):
+            datalake_compute_environment = compute_environment
+            logger.info("Datalake Compute Environment: %s", datalake_compute_environment)
+            break
+
+    return datalake_compute_environment
+
+
+def launch_compute_environment_resources(batch_client, datalake_compute_environment):
+    """
+    Trigger immediate launch of computing resources. This will significantly speed up Batch jobs
+    execution.
+    """
+    if (
+        int(datalake_compute_environment["computeResources"]["minvCpus"]) < 1
+        or int(datalake_compute_environment["computeResources"]["desiredvCpus"]) < 1
+    ):
+        batch_client.update_compute_environment(
+            computeEnvironment=datalake_compute_environment["computeEnvironmentName"],
+            computeResources={"minvCpus": 1, "desiredvCpus": 1},
+        )
+
+
+def get_state_machine(stepfunctions_client):
+    state_machines_detection_response = stepfunctions_client.list_state_machines()
+
+    # make sure there are no more records in next pages
+    assert state_machines_detection_response.get("nextToken", None) is None
+
+    for state_machine in state_machines_detection_response["stateMachines"]:
+
+        tags_detection_response = stepfunctions_client.list_tags_for_resource(
+            resourceArn=state_machine["stateMachineArn"]
+        )
+
+        # reformat returned array of tags to dictionary
+        state_machine_tags = {tag["key"]: tag["value"] for tag in tags_detection_response["tags"]}
+
+        if (
+            state_machine_tags[ENVIRONMENT_TAG_KEY] == ENVIRONMENT_TAG_VAL
+            and state_machine_tags.get(APPLICATION_TAG_KEY, None) == APPLICATION_TAG_VAL
+        ):
+            datalake_state_machine = state_machine
+            logger.info("Datalake State Machine: %s", datalake_state_machine)
+            break
+
+    return datalake_state_machine
+
+
+@mark.timeout(1200)
+@mark.infrastructure
+def test_should_successfully_run_dataset_version_creation_process(
+    batch_client, stepfunctions_client
+):
+
+    test_data_bucket = "s3://linz-geospatial-data-lake-test-data"
```

Comment:
This looks like reusable information; should it maybe be a constant in utils.py?

Reply:
It is in separate variable only because otherwise I have too long line below (and Pylint wouldn't pass). And I guess this will need to be replaced by S3Object .

Reply:
`TEST_DATA_BUCKET` shouldn't be any longer?

Reply:
S3Object used in 8e466364d2d6f8b5d86e5be609d31174704e85d0